File: Cap 10/Topico 10.5/Troca Valores V2/tricaval2.c
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 10 - Topico 10.5
 * 
 * Programa ..: TROCA: COM PROCEDIMENTO
 */

#include <stdio.h>

int A, B;

  void TROCA(void)
  {
    int X = A;
    A = B;
    B = X;
  }

int main(void)
{
  printf("TROCA DE VALORES - V2\n\n");
  printf("Entre valor para variavel <A>: ");
  scanf("%d", &A);
  printf("Entre valor para variavel <B>: ");
  scanf("%d", &B);
  TROCA();
  printf("\n");
  printf("Os valores trocados sao:\n\n");
  printf("<A> = %d\n", A);
  printf("<B> = %d\n", B);
  return 0;
}

File: Cap 10/Topico 10.5/Troca Valores V2/tricaval2.dart
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 10 - Topico 10.5
 * 
 * Programa ..: TROCA: COM PROCEDIMENTO
 * 
 */
 
import 'dart:io';

int A = 0, B = 0;

  void troca() {
    int X = A;
    A = B;
    B = X;
  }

void main() {
  print('TROCA DE VALORES - V2');
  print('');
  stdout.write('Entre valor para variavel <A>: ');
  A = int.parse(stdin.readLineSync()!);
  stdout.write('Entre valor para variavel <B>: ');
  B = int.parse(stdin.readLineSync()!);
  troca();
  print('');
  print('Os valores trocados sao:');
  print('');
  print('<A> = $A');
  print('<B> = $B');
}
